server/src/Irony.cpp

```diff
@@ -258,17 +259,29 @@ void Irony::complete(const std::string &file,
       unsigned annotationStart = 0;
       bool typedTextSet = false;
 
-      if (availability == CXAvailability_NotAccessible ||
-          availability == CXAvailability_NotAvailable) {
-        continue;
-      }
-
       typedtext.clear();
       brief.clear();
       resultType.clear();
       prototype.clear();
       postCompCar.clear();
       postCompCdr.clear();
+      available.clear();
+
+      switch (availability) {
```

Comment:
For me the special case here is the case don't transform to string and ignore.
I would put it first and add a small comment:

      case CXAvailability_NotAvailable:
          // no benefits to expose this to Elisp for now
          continue;